src/assets/css/material/_theme.scss

```diff
@@ -358,8 +358,23 @@
     background-color: map.get($colors, primary);
   }
 
-  .mat-checkbox-disabled .mat-checkbox-inner-container .mat-checkbox-background {
-    background-color: map.get($colors, checkbox-disabled);
+  .mat-checkbox-disabled {
```

Comment:
Change looks good IMO, i tried following sceanrios

- checked + disabled
- checked + not-disabled
- unchecked + disabled
- unchecked + not-disabled

and it looks as per UI improvement 👍🏽